src/buckets/components/createBucketForm/UpdateBucketOverlay.tsx

```diff
@@ -131,10 +156,58 @@ const UpdateBucketOverlay: FunctionComponent<Props> = ({
     }
   }
 
+  const isValid = () => {
+    // are there measurement schemas?
+    const haveSchemas =
+      !!newMeasurementSchemaRequests && !!newMeasurementSchemaRequests.length
+
+    if (!haveSchemas) {
+      // no schemas, nothing to validate, so everything is fine
+      return true
+    }
+    // if so, are they all valid?
+    const reducer = (previousValue, currentValue) =>
+      previousValue && currentValue.valid
+
+    const alltrue = newMeasurementSchemaRequests.reduce(reducer, true)
```

Comment:
the [`some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) method is made for this kind of calculation.

```ts
const allTrue = !newMeasurementSchemaRequests.some(schema => schema.valid === false)

Reply:
thanks!  I keep forgotting about the some method

Reply:
actually, switching to use 'every' because negative tests are to be avoided if possible.
but yeah, using optimized array methods

` const alltrue = newMeasurementSchemaRequests.every(schema => schema.valid)`

Reply:
even better! I _knew_ there was a cleaner way to do it than negating `some`